packages/logger/tests/unit/logEvent.test.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * Logger log event tests
+ *
+ * @group unit/logger/logger/logEvent
+ */
+import context from '@aws-lambda-powertools/testing-utils/context';
+import { beforeEach, describe, expect, it, jest } from '@jest/globals';
+import middy from '@middy/core';
+import type { Context } from 'aws-lambda';
+import { Logger } from '../../src/Logger.js';
+import { injectLambdaContext } from '../../src/middleware/middy.js';
+
+const event = {
+  foo: 'bar',
+};
+
+const logSpy = jest.spyOn(console, 'info');
+
+describe('Log event', () => {
+  const ENVIRONMENT_VARIABLES = process.env;
+
+  beforeEach(() => {
+    process.env = {
+      ...ENVIRONMENT_VARIABLES,
+      POWERTOOLS_LOGGER_LOG_EVENT: 'true',
+      POWERTOOLS_DEV: 'true',
+    };
+    jest.resetAllMocks();
+  });
+
+  it('logs the event with the correct log level and message', () => {
+    // Prepare
+    const logger = new Logger();
+
+    // Act
+    logger.logEventIfEnabled(event);
+
+    // Assess
+    expect(logSpy).toHaveBeenCalledTimes(1);
```

Comment:
Need this?

Reply:
Strictly speaking it's not needed, but it aims at giving a better error message in case of failure.

In the next line, we access the content of the first call via `logSpy.mock.calls[0][0]`, which in case of the spy not getting called would fail with an error similar to "unable to access index 0 of undefined" (I'm paraphrasing the error message for clarity).

I found the error message a bit confusing and I would rather it failing earlier at the `expect(logSpy).toHaveBeenCalledTimes(1);` which gives me a better error of "spy hasn't been called at all" (again, paraphrasing).

This is a lot of repetition though, every time I need to check the content of a call I need to first check that it was called, then parse it (next line), etc. I am already planning on creating a custom assertion in the future, which would allow me to replace both lines with this `expect(logSpy).toHaveLoggedObjectContaining({ foo: 'bar' })` but this would be test-runner specific and I'm waiting to migrate this package to vitest before actually doing it.

Reply:
Aaa ok, I see! You are trying to avoid generic errors like undefined/key_errors and focusing on having the utility response, that way if at least 1 call is not made you know that something failed and not a generic key error. Thanks for explaining.